samza-api/src/main/java/org/apache/samza/table/retry/TableRetryPolicy.java

```diff
@@ -256,4 +266,30 @@ public BackoffType getBackoffType() {
   public RetryPredicate getRetryPredicate() {
     return retryPredicate;
   }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public Map<String, String> toConfig(Config jobConfig, Config tableConfig) {
+    final Gson gson = new GsonBuilder().excludeFieldsWithModifiers(Modifier.TRANSIENT, Modifier.STATIC).create();
+    return Collections.singletonMap(JavaTableConfig.buildKey(tableId, this.getClass().getCanonicalName()),
```

Comment:
Right. After adding `io.read.retry.policy` as part of the tableId, I should remove this `ClassName`.

Reply:
But if we don't keep the ClassName, this key will be the same as the key we have been using to (de)serialize the TableRetryPolicy. And thus it will override the existing key. I suggest we use a short ClassName here without the namespace.